internal/testutil/util.go

```diff
@@ -241,25 +243,40 @@ func CheckStatusCodeOK(ctx context.Context, t *testing.T, httpClient *http.Clien
 			t.Logf("Got %d response code, expected %d: %v. Retrying.", resp.StatusCode, http.StatusOK, resp)
 
 			return false
-		case http.StatusOK:
-			data, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
-				t.Fatalf("Reading response body: %v", err)
+		default:
+			if failCondition(resp.StatusCode) {
+				t.Fatalf("Unexpected response code %d", resp.StatusCode)
 			}
 
-			body = data
+			if resp.StatusCode == http.StatusOK {
+				data, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					t.Fatalf("Reading response body: %v", err)
+				}
+
+				body = data
+			}
```

Comment:
The `switch` now has only one case statement, I think we could change it a bit to include `failCondition` or get rid of it either. Maybe something like this:

```go
switch {
case resp.StatusCode == http.StatusForbidden, resp.StatusCode == http.StatusUnauthorized:
// ...
case failCondition(resp.StatusCode):
// ...
}

// <read body block here>
```

WDYT?

Reply:
Updated and simplified a bit as well by returning always the body regardless of the status code, thanks. Check 7fc46b2386ebe64739b311a40ba4bd4e5fd97882.